copycat/api/src/main/java/org/apache/kafka/copycat/data/SchemaBuilder.java

```diff
@@ -346,7 +347,7 @@ public Schema valueSchema() {
      * @return the {@link Schema}
      */
     public Schema build() {
-        return new CopycatSchema(type, isOptional(), defaultValue, name, version, doc, fields, keySchema, valueSchema);
+        return new CopycatSchema(type, isOptional(), defaultValue, name, version, doc, fields == null ? null : Collections.unmodifiableList(fields), keySchema, valueSchema);
```

Comment:
that is how we do schema-less? a schema without fields? or is this for a different use-case?